python/lsst/atmospec/spectraction.py

```diff
@@ -133,22 +138,38 @@ def dumpParameters():
             if not item.startswith("__"):
                 print(item, getattr(parameters, item))
 
-    def spectractorImageFromLsstExposure(self, exp, *, target_label='', disperser_label='', filter_label=''):
+    def debugPrintTargetCentroidValue(self, image):
+        x, y = image.target_guess
+        self.log.debug(f"Image shape = {image.data.shape}")
+        self.log.debug(f"x, y = {x}, {y}")
+        x = int(np.round(x))
+        y = int(np.round(y))
+        self.log.debug(f"Value at {x}, {y} = {image.data[y, x]}")
+
+    def spectractorImageFromLsstExposure(self, exp, xpos, ypos, *, target_label='',
+                                         disperser_label='', filter_label=''):
         """Construct a Spectractor Image object from LSST objects.
 
         Internally we try to use functions that calculate things and return
         them and set the values using the return rather than modifying the
         object in place where possible. Where this is not possible the methods
         are labeled _setSomething().
         """
+        # TODO: passing exact centroids seems to be causing a serious
+        # and non-obvious problem!
+        # this needs fixing for several reasons, mostly because if we have a
+        # known good centroid then we want to skip the refitting entirely
+        xpos = int(np.round(xpos))
+        ypos = int(np.round(ypos))
+        self.log.debug(f'DM value at centroid: {exp.image.array[ypos, xpos]}\n')
+
+        # make a blank image, with the filter/disperser set
         image = Image(file_name='', target_label=target_label, disperser_label=disperser_label,
                       filter_label=filter_label)
 
         vi = exp.getInfo().getVisitInfo()
         rotAngle = vi.getBoresightRotAngle().asDegrees()
-        # line below correct if not rotating 90 XXX remove this once resolved
-        # parameters.OBS_CAMERA_ROTATION = 180 - (rotAngle % 360)
-        parameters.OBS_CAMERA_ROTATION = 90 - (rotAngle % 360)
+        parameters.OBS_CAMERA_ROTATION = 270 - (rotAngle % 360)
```

Comment:
Is this a time dependent parameter, such that only data after date D works with the new version?

Reply:
No, it shouldn't be - anything like that should be being fixed up in the LATISS translator and/or yaml fixup files - this is just a difference in geometric definitions.